src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php

```diff
@@ -75,7 +81,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         foreach ($pools as $id => $pool) {
             $io->comment(sprintf('Clearing cache pool: <info>%s</info>', $id));
-            $pool->clear();
+
+            if ($pool instanceof CacheItemPoolInterface) {
+                $pool->clear();
+            } else {
+                $globalClearer->clearPool($id);
```

Comment:
Now I understand why you needed hasPool :)
This can throw in the middle of a clean because of a missing pool.
Then I propose to replace clearPool($pool) by `PsrCacheClearer::getPools()` and handle validation with the resulting array.
WDYT?